### PR TITLE
Reproduce Amazon.QLDBSession.Model.BadRequestException : Digests don't match

### PR DESCRIPTION
**Pull request by adrianiftode**
*Description of changes:*

This draft reproduces an issue on .NET Framework 4.6.1, one of the .NET Standard implementations and it is not meant to merge.

The issue is when an ION document with a struct having more than one field is inserted into a table, then we get the following exception `Amazon.QLDBSession.Model.BadRequestException : Digests don't match`

`Execute_InsertDocumentWithMultipleFields_DocumentIsInserted` is the test that reproduces this issue. It passes on .NET Core 2.0, 2.1 and 3.1, but not on NET Framework 4.6.1

Associated stack trace
```
  Error Message:                                                                                                                                                                                                                              
   Test method Amazon.QLDB.Driver.IntegrationTests.StatementExecutionTests.Execute_InsertDocumentWithMultipleFields_DocumentIsInserted threw exception:                                                                                       
Amazon.QLDBSession.Model.BadRequestException: Digests don't match ---> Amazon.Runtime.Internal.HttpErrorResponseException: The remote server returned an error: (400) Bad Request. ---> System.Net.WebException: The remote server returned an
 error: (400) Bad Request.                                                                                                                                                                                                                    
  Stack Trace:                                                                                                                                                                                                                                
      at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)                                                                                                                                                                   
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)                                                                      
--- End of stack trace from previous location where exception was thrown ---                                                                                                                                                                  
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)                                                                                                                                                               
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)                                                                                                                                          
   at Amazon.Runtime.Internal.HttpRequest.<GetResponseAsync>d__16.MoveNext()                                                                                                                                                                  
--- End of inner exception stack trace ---                                                                                                                                                                                                    
    at Amazon.Runtime.Internal.HttpRequest.<GetResponseAsync>d__16.MoveNext()                                                                                                                                                                 
--- End of stack trace from previous location where exception was thrown ---                                                                                                                                                                  
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)                                                                                                                                                               
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)                                                                                                                                          
   at Amazon.Runtime.Internal.HttpHandler`1.<InvokeAsync>d__9`1.MoveNext()                                                                                                                                                                    
--- End of stack trace from previous location where exception was thrown ---                                                                                                                                                                  
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)                                                                                                                                                               
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)                                                                                                                                          
   at Amazon.Runtime.Internal.Unmarshaller.<InvokeAsync>d__3`1.MoveNext()                                                                                                                                                                     
--- End of stack trace from previous location where exception was thrown ---                                                                                                                                                                  
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)                                                                                                                                                               
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)                                                                                                                                          
   at Amazon.Runtime.Internal.ErrorHandler.<InvokeAsync>d__5`1.MoveNext()                                                                                                                                                                     
--- End of inner exception stack trace ---                                                                                                                                                                                                    
    at Amazon.QLDB.Driver.RetryHandler.RetriableExecute[T](Func`1 func, RetryPolicy retryPolicy, Action recoverAction, Action`1 retryAction) in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver\retry\RetryHandler.cs:line 66  
   at Amazon.QLDB.Driver.SessionPool.Execute[T](Func`2 func, RetryPolicy retryPolicy, Action`1 retryAction) in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver\session\SessionPool.cs:line 64                                  
   at Amazon.QLDB.Driver.QldbDriver.Execute[T](Func`2 func, RetryPolicy retryPolicy, Action`1 retryAction) in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver\driver\QldbDriver.cs:line 223                                    
   at Amazon.QLDB.Driver.QldbDriver.Execute[T](Func`2 func, RetryPolicy retryPolicy) in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver\driver\QldbDriver.cs:line 197                                                          
   at Amazon.QLDB.Driver.QldbDriver.Execute[T](Func`2 func) in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver\driver\QldbDriver.cs:line 144                                                                                   
   at Amazon.QLDB.Driver.IntegrationTests.StatementExecutionTests.Execute_InsertDocumentWithMultipleFields_DocumentIsInserted() in D:\adi\opensource\amazon-qldb-driver-dotnet\Amazon.QLDB.Driver.IntegrationTests\StatementExecutionTests.cs:
line 286                                                                                                                                                                                                                                      
```



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yoroto** [comment]
Hi @adrianiftode 

Thank for reporting this! Will look into it this week.

File: Amazon.QLDB.Driver.Tests/Amazon.QLDB.Driver.Tests.csproj [hunk 1]
```diff
@@ -1,16 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp2.0;netcoreapp2.1;netcoreapp3.1;net461</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (yoroto) [reply Amazon.QLDB.Driver.Tests/Amazon.QLDB.Driver.Tests.csproj]: Thanks! LGTM. Can you merge the latest retry policy change please?

[the remark this answers is not in the document]
Comment (yoroto) [reply Amazon.QLDB.Driver.Tests/Amazon.QLDB.Driver.Tests.csproj]: BTW, the Ion Hash bug causing this issue has been fixed. I will make change to bring it in later.

[the remark this answers is not in the document]
Comment (adrianiftode) [reply Amazon.QLDB.Driver.Tests/Amazon.QLDB.Driver.Tests.csproj]: Updated

**Review by yoroto — APPROVED** [review]

**Pull request merged**